python/fledge/services/core/api/utils.py

```diff
@@ -27,6 +27,9 @@ def get_plugin_info(name, dir):
         out, err = p.communicate()
         res = out.decode("utf-8")
         jdoc = json.loads(res)
+    except json.decoder.JSONDecodeError as err:
+        _logger.error("Failed to parse JSON data returned from the plugin information of {}, {} line {} column {}".format(name, err.msg, err.lineno, err.colno))
```

Comment:
Use multiline stacktrace logger style.

`_logger.error(err, "Failed to parse JSON data returned from the plugin information of {}.".format(name))`

Reply:
The concept here is to give the user a very clear error message they can pick up on rather than expect them to look at something more general and work out what went wrong.

Reply:
Sure. Previously, if we have some invalid JSON for a plugin it raises below
`ERROR: logger: fledge.services.core.api.utils: json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 2870 (char 2869)`

And with the given change
`ERROR: logger: fledge.services.core.api.utils: Failed to parse JSON data returned from the plugin information of PLUGIN_NAME, Expecting ',' delimiter line 1 column 2870`

So the trailing message still seems to be more developer related logger.